cmd/get.go

```diff
@@ -64,18 +59,14 @@ to decrypt it securely.`,
 			filedest = filedest + "/" + objectComponents[len(objectComponents)-1]
 		}
 
-		sess := viper.Get("ClientSess").(*session.Session)
-		svc := s3crypto.NewDecryptionClient(sess)
-		svc.CEKRegistry[crosscrypto.AESCBCPKCS5Padding] = crosscrypto.NewAESCBCContentCipher
-
-		decryptionclient := Get{
-			Client:          svc.S3Client,
+		getobj := &GetObject{
 			Bucket:          s3bucket,
 			Key:             s3object,
 			FileDestination: filedest,
 		}
+		decryptionclient := NewDecryptionClient()
 
-		content, err := decryptionclient.getS3Cse()
+		content, err := GetS3Cse(getobj, decryptionclient)
```

Comment:
So I'm not sure how I feel about this implementation:
- What's the benefit of removing composition? Seems like that would be more cohesive, and maybe more portable. Unless this is necessary for how your tests are setup?
 i.e. `func (g *GetObject) GetS3Cs3(decryptionclient S3DecryptionClientAPI) (io.ReadCloser, error) {}`
- Also, you're handling the error in GetS3Cse and then returning it but not actually utilizing it after you return it. So maybe just don't return the error, or wait to handle it after the call?

Reply:
Resolving this conversation with 8835e77 change